### PR TITLE
Frontend profile edit

### LINKED ISSUE #85: Profile edit page
- [ ] Form to edit user details
- [ ] Link to Gravatar to set profile picture
- [ ] Tests

### PR DESCRIPTION
**Pull request by JuhoBjn**
Add a page to edit user profiles.

Closes #85

### CODE REVIEW

File: frontend/src/pages/user-profile/UserProfile.jsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    const response = await sendFriendRequest(
+      authContext.id,
+      user.id,
+      authContext.token
+    );
+    if (response) {
+      console.log(response.message);
+      setFriendRequestSent(true);
+    } else {
+      console.log("Failed to send friend request.");
```
Comment (Jypaa) [inline frontend/src/pages/user-profile/UserProfile.jsx:29]: Console.log

File: frontend/src/pages/user-profile/UserProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  const [friendRequestSent, setFriendRequestSent] = useState(false);
+
+  const sendFriendRequestHandler = async () => {
+    const response = await sendFriendRequest(
+      authContext.id,
+      user.id,
+      authContext.token
+    );
+    if (response) {
+      console.log(response.message);
```
Comment (Jypaa) [inline frontend/src/pages/user-profile/UserProfile.jsx:26]: Console.log

**Review by Jypaa — APPROVED** [review]

File: frontend/src/pages/edit-profile/EditProfile.jsx [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+      navigate(`/user/${userProfile.id}`);
+    } else {
+      console.log("Profile update failed");
+    }
+  };
+
+  return (
+    <div className="edit-profile-page">
+      <div className="edit-profile-form-container">
+        <h2>Profile editing</h2>
```
Comment (vaasuu) [inline frontend/src/pages/edit-profile/EditProfile.jsx:58]: There is no error messages shown to the user when the request fails. The API gives error messages that the fields cant be empty -> should show the errors to the user and/or make the form fields `required` so they cant be empty.

**Review by vaasuu — COMMENTED** [review]
Looks good enough for me. I left a comment about user edit form error messages not being shown.

**Review by vaasuu — APPROVED** [review]

File: frontend/src/pages/user-profile/UserProfile.jsx (hunk 1 above)
Comment (JuhoBjn) [reply frontend/src/pages/user-profile/UserProfile.jsx]: Resolved this in commit https://github.com/JuhoBjn/online-store/pull/102/commits/f58f504926e0ae1ead8e4509287280a9b8712bee

File: frontend/src/pages/edit-profile/EditProfile.jsx [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+          <label htmlFor="country-input">Country</label>
+          <select
+            name="country"
+            className="country-dropdown"
+            id="country-input"
+            data-testid="country-input"
+            value={userProfile.country}
+            onChange={() => updateUserProfileState(event)}
+          >
+            <option value="af">Afghanistan</option>
```
[the remark this answers is not in the document]
Comment (JuhoBjn) [reply frontend/src/pages/edit-profile/EditProfile.jsx:155]: I'll create a ticket to refactor this after the testing with IB students.

File: frontend/src/pages/edit-profile/EditProfile.jsx (hunk 3 above)
Comment (JuhoBjn) [reply frontend/src/pages/edit-profile/EditProfile.jsx:58]: I'll make the fields required for now. Perhaps I'll add an error message later.

**Pull request merged**